src/pages/playoffSeries.js

```diff
@@ -0,0 +1,255 @@
+  const id = req.params.id;
+
+  const season = await _season.getSeason(season_id);
+  const teams = await _team.getTeams(season.id);
+  const playoffSeries = await _series.getSeries({
+    season_id: season_id,
+    series_id: id,
+    is_playoff: true
+  });
+
+  const series = playoffSeries[0];
+  series.home = {};
+  series.home.id = series.home_team_id;
+  series.home.name = series.home_team_name ? series.home_team_name : 'Undecided';
+  series.home.logo = series.home_team_logo;
+  series.home.points = series.home_points;
+  series.away = {};
+  series.away.id = series.away_team_id;
+  series.away.name = series.away_team_name ? series.away_team_name : 'Undecided';
+  series.away.logo = series.away_team_logo;
+  series.away.points = series.away_points;
+
+  const html = templates.playoffSeries.edit({
+    user: req.user,
+    verb: 'Edit',
+    season: season,
+    teams: teams,
+    series: series,
+    csrfToken: req.csrfToken()
+  });
+
+  res.send(html);
+}
+
+function post(_series, _team, req, res) {
+  if (!req.user || !req.user.isAdmin) {
+    res.sendStatus(403);
+    return;
+  }
+
+  const season_id = req.body.season_id;
+  const id = req.body.id ? req.body.id : shortid.generate();
+  const series = req.body;
+  series.id = id;
+
+  if (series.home_team_id === '') {
+    series.home_team_id = null;
+  }
+
+  if (series.away_team_id === '') {
+    series.away_team_id = null;
+  }
+
+  const seriesUrl = series.series_url;
+  if (!seriesUrl) {
+    series.series_url = null;
+  }
+
+  if (series.home_points === '') {
+    series.home_points = null;
+  }
+
+  if (series.away_points === '') {
+    series.away_points = null;
+  }
+
+  if (!series.home_team_id && !series.away_team_id) {
+    res.redirect('/seasons/' + season_id + '/playoff-series');
+  }
+
+  series.is_playoff = true;
+
+  _series.saveSeries(series).then(() => {
+    res.redirect('/seasons/' + season_id + '/playoff-series');
+  }).catch(err => {
+    console.error(err)
+    res.sendStatus(500);
+  })
+}
+
+function remove(_series, req, res) {
+  if (!req.user || !req.user.isAdmin) {
+    res.sendStatus(403);
+    return;
+  }
+
+  const season_id = req.body.season_id
+  const id = req.body.id
+
+  _series.deleteSeries(id).then(() => {
+    res.redirect('/seasons/' + season_id + '/playoff-series');
+  }).catch(err => {
+    console.error(err)
+    res.sendStatus(500);
+  })
+}
+
+async function bracket(templates, _season, _team, _series, _pairings, req, res) {
+  const season_id = req.params.season_id
+
+  const season = await _season.getSeason(season_id);
+  const series = await _series.getSeries({
+    season_id: season_id,
+    is_playoff: true
+  });
+
+  const roundOne = series.filter((matchup) => {
+    return matchup.round === 1;
+  });
+
+  const numberOfMatchups = roundOne.length * 2;
+  const numRounds = Math.ceil(Math.log2(numberOfMatchups));
+
+  let remainingSeries = series.slice(roundOne.length - 1);
+  let currentRoundNum = 2;
```

Comment:
Is this supposed to be hard-coded?

Reply:
Yea, we start processing the rounds at the second.
I'm doing some tweaks that kabo mentioned above, I'll add some more documentation about this processing at the same time